library/func_misc.py

```diff
@@ -153,3 +153,53 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Join(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        li = r.PosList()
+
+        delim = ''
+        if len(pos_args): # reader has a reference
+            delim = r.PosStr()
```

Comment:
OK actually this feels like we need to update our API for positional defaults

like `r.PosStr(default='')` or something